packages/zipkin/test/batch-recorder.test.js

```diff
@@ -486,4 +490,15 @@ describe('Batch Recorder - integration test', () => {
 
     expect(spans).to.be.empty; // eslint-disable-line no-unused-expressions
   });
+
+  it('should not log span that is not in the reference', () => {
+    const span = {
+      delegate: {
+        timestamp: 1,
+        setLocalEndpoint: () => {}
+      }
+    };
+    recorder._writeSpan(rootId, span);
+    expect(spans).to.be.empty; // eslint-disable-line no-unused-expressions
```

Comment:
This test makes sure this is right: https://github.com/openzipkin/zipkin-js/pull/449/files#diff-503c850b99353bea68792242ff75f9edR114